stores/sql/sqlite/main.go

```diff
@@ -731,12 +731,12 @@ func (tx *MainDatabaseTx) PruneHostSectors(ctx context.Context, limit int64) (in
 	res, err := tx.Exec(ctx, `DELETE FROM host_sectors
 WHERE rowid IN (
 	SELECT rowid
-	FROM host_sectors hs
-	WHERE NOT EXISTS (
-		SELECT 1
+	FROM host_sectors
```

Comment:
To keep things simple I think we can use the same query we use for mysql here? Might also be slightly faster since we already need to select all the rows to delete to get their row id.

Reply:
It's very important we can `LIMIT` the number of sectors we prune in a single batch. MySQL allows `DELETE ... LIMIT` but `SQLite` doesn't. This SQLite query is the fastest of all the ones I've tried, that's where `BenchmarkPruneHostSectors` came in handy.

Reply:
Ok that makes sense. Do we have tests that verify pruning is sane and only deletes up to `limit` sectors of which all of them are ok to delete?
I'm afraid that with duplicate host ids in the `host_sectors` we aren't actually using the whole limit.